addon/ActivityStreams.js

```diff
@@ -95,6 +95,10 @@ function ActivityStreams(metadataStore, tabTracker, telemetrySender, options = {
     fetchNewMetadata: (links, type) => {
       const event = this._tabTracker.generateEvent({source: type});
       return this._previewProvider.asyncSaveLinks(links, event);
+    },
+    fetchNewMetadataLocally: (links, type) => {
+      const event = this._tabTracker.generateEvent({source: type});
+      return this._pageScraper.asyncFetchLinks(links, event);
```

Comment:
Okay I know this is a little annoying, but can you change this so that here we loop over links like so:


    for (let link of links) {
        const event = this._tabTracker.generateEvent({source: type});
        this._pageScraper.asyncFetchLink(link, event);
    }

And all the methods just deal with a single link, that way we have a single telemetry event for each link we're fetching and all the methods only deal with a single link?